libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -213,8 +220,96 @@ void AP_MotorsUGV::set_throttle(float throttle)
         return;
     }
 
+    
     // check throttle is between -_throttle_max and  +_throttle_max
-    _throttle = constrain_float(throttle, -_throttle_max, _throttle_max);
+    float throttle_chk = constrain_float(throttle, -_throttle_max, _throttle_max);
```

Comment:
I don't think you need this extra `throttle_chk` var. If you leave it as `_throttle` you can remove the sets on lines 229, 303 and 307.